metadata/src/test/java/org/apache/kafka/controller/PartitionChangeBuilderTest.java

```diff
@@ -217,6 +228,16 @@ public void testTriggerLeaderEpochBumpIfNeeded() {
             new PartitionChangeRecord(),
             1
         );
+
+        // Shrinking the ISR while in ZK migration mode does increase the leader epoch
```

Comment:
Add a reference to KAFKA-15109. Also a reason: I guess it's because the ZK brokers need this information.